### PR TITLE
[RDBMS] Add Private DNS zone for every database server created with private network access

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**

- PostgreSQL server created with private network access (Vnet and subnet) will be using private DNS zone from GA.
- az postgres flexible-server create/restore commands are updated. 
- Enable database name for create command
- Enable check server name availability for create command

**Testing Guide**

**History Notes**

[RDBMS] BREAKING CHANGE: az mysql flexible-server create command parameter --storage-size default value is changed from 10 to 32. 
[RDBMS] az postgres flexible-server create command added --private-dns-zone parameter for creating server with private access. 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by DaeunYim** [comment]
Create command without Private DNS zone support will not be supported post GA of flexible server.  Is there any way we can guide the customers to upgrade the CLI version?

**Comment by yonzhan** [comment]
@Juliehzl @evelyn-ys to help with

**Comment by Juliehzl** [comment]
> Create command without Private DNS zone support will not be supported post GA of flexible server. Is there any way we can guide the customers to upgrade the CLI version?

@DaeunYim We cannot change behavior for existing Azure CLI users. As you mentioned, without Private DNS zone will not be support. Does it mean without private DNS parameter, previous command will fail？

**Comment by DaeunYim** [comment]
> users. As you mentioned, without Private DNS zone will not be support. Does it mean without private DNS para

Yes post GA it will eventually be deprecated. Let me communicate with the backend team to give a good actionable error on this..

**Comment by DaeunYim** [comment]
This PR needs new Python SDK release. I have checked the tests are passing with the locally generated SDK. Let me work with the SDK release vendor and add the version once it's done,,

**Comment by DaeunYim** [comment]
Confirmed all failing tests are due to fault in latest Python SDK version. Let me work on it as soon as the new SDK is out,,

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -269,7 +269,7 @@ def _flexible_server_params(command_group):
                            help='Compute tier of the server. Accepted values: Burstable, GeneralPurpose, Memory Optimized ')
                 c.argument('sku_name', default='Standard_B1ms', options_list=['--sku-name'],
                            help='The name of the compute SKU. Follows the convention Standard_{VM name}. Examples: Standard_B1ms, Standard_E16ds_v4 ')
-                c.argument('storage_mb', default='10', options_list=['--storage-size'], type=int,
+                c.argument('storage_mb', default='32', options_list=['--storage-size'], type=int,
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:272]: default value change is breaking change to customer. please add in release note.

**Comment by Juliehzl** [comment]
> > users. As you mentioned, without Private DNS zone will not be support. Does it mean without private DNS para
> 
> Yes post GA it will eventually be deprecated. Let me communicate with the backend team to give a good actionable error on this..

Hi @DaeunYim, as there are thousands of command invocation for `flexible-server create`, typically such breaking change is not allowed and we need to follow Azure CLI breaking change process. Azure CLI allows breaking change twice a year and the latest one is for build. Could we make the breaking change in S187 release?

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 1 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:272]: Added it

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**